Day 3/solution2/src/main.rs

```diff
@@ -1,68 +1,23 @@
-use std::fs::File;
-use std::io::{self, BufRead};
-use std::path::Path;
-// use std::slice;
+use std::fs;
 
 fn main() {
-    // let mut counts : Vec<Vec<u32>> = vec![vec![0, 2]; 12];
-    let count: Vec<u32> = {
-        let lines: io::Lines<_> = read_lines("../input.txt").unwrap();
-        let lines_vector: Vec<String> = lines.map(|line| line.unwrap()).collect::<Vec<String>>();
-        let lines_count_vector: Vec<Vec<u32>> = lines_vector
-            .into_iter()
-            .map(|line| line.char_count('1'))
-            .collect();
-        let counts = lines_count_vector
-            .into_iter()
-            .reduce(|a, b| a.add(b))
-            .unwrap();
-        counts
-            .iter()
-            .map(|count| if *count >= 500 { 1 } else { 0 })
-            .collect()
```

Comment:
You did make really solid use of iterator functions here, I really like how you went about getting the info you needed. It can definitely be harder when you don't know the types of stuff as it's getting passed through the chain, but Rust Analyzer should do the heavy lifting here by hinting types in grey :+1: